### ISSUE TITLE
windows dev support

### ISSUE THREAD

**Issue by svenevs** [svenevs/exhale#131]
> I had trouble running exhale tests on windows because of the lxml pkg trying to pip install without access to VS's cl.exe disappointed.
>
> https://github.com/michaeljones/breathe/pull/766#issuecomment-992034888

@2bndy5 was the issue that you didn't have visual studio (provides `cl.exe`) installed?  They do [release wheels for windows](https://pypi.org/project/lxml/#files), but otherwise yeah it's gotta compile the underlying libraries from source.  Otherwise, maybe you are on python 3.10?  Doesn't look like a windows binary was released for that yet.

Just trying to double check what the exact problem is, we've got 3.10 running in CI here now.  What did you do to resolve the issue?  (Or was it never resolved? hehe)

**Comment by 2bndy5** [comment]
> Just trying to double check what the exact problem is

I tried to reproduce this, but I couldn't. `tox -e py -- --cov-report xml:coverage.xml --cov` runs fine on windows now.

lxml must've fixed this recently. I suspect [this PR](https://github.com/lxml/lxml/pull/326) is the victorious achievement in that regard. Notice, the latest lxml release (v4.7) is 5 hours old. Whereas v4.6.5 (the last stable release) did not have wheels built for windows 🤓 

> What did you do to resolve the issue? 

Previously I had to install lxml on windows via a 3<sup>rd</sup> party distributed wheel (as recommended by the pkg's docs). However, I didn't see an easy way to do this with tox because it would've likely meant editing the _tox.ini_ or something.

For people working with older versions of the lxml pkg on windows, I suspect they had to do the same thing (or `pip install lxml` from a VS dev prompt which would expose the needed `cl.exe`). For the record, I had to do 3<sup>rd</sup> party wheel solution a couple years ago when I last tried using exhale.

> we've got 3.10 running in CI here now

I see that! Awesome! Any thoughts as to optimizing the workflow's layout (as [I previously suggested using the matrix better](https://github.com/svenevs/exhale/pull/114#pullrequestreview-793529121))?

**Comment by svenevs** [comment]
Oh haha wow what timing! :upside_down_face: Ok, interesting.  So I wonder if we can reproduce it by choosing a much older version.  In current `master` the `tox` setup should be functional but I was reworking the setup stuff in #130 so if not that branch is better for the time being (blocked by the breathe thing temporarily).

So now the question is which one did you install manually by source?  Probably `v4.6.4`.  In `tox.ini` in the `[testenv]` set the `deps = ...` with an `lxml==4.6.4`, `tox -e py` should reproduce your issue.

1. I'm not super familiar with who controls the build isolation, `tox` or `pip` or both, but if you added a `whitelist_externals = cl.exe` to `[testenv]` section (example in `[testenv:clean]` which you can't even run on windows xD), does that get you past it?  The theory is that if behind the scenes `lxml` has access to the compiler, I believe it will succeed in building from source (it'll take a bit longer, but it's not a very large project where cxx is concerned).
2. After checking (1), delete the `.tox` directory so we can start fresh.  Since you did the manual install, you should be able to run `tox --sitepackages -e py` which should skip trying to create virtual environments.  The implication being that if you install the exhale requirements and testing requirements (everything else in that `deps` section), you'll be able to run tests again.

(1) may be worth adding if it is successful, or documenting (2).

> Any thoughts as to optimizing the workflow's layout

Absolutely :slightly_smiling_face:  At some point in that PR I had a matrix version but got rid of it for some reason.  Going through misc repo cleanup right now though and that's definitely slated!

**Comment by 2bndy5** [comment]
> which one did you install manually by source?

Ironically, I just deleted the downloaded wheel a few minutes ago after realizing that lxml now has wheels for windows. My recycling bin says the wheel was for v4.6.3

> I'm not super familiar with who controls the build isolation, tox or pip or both

I'm pretty sure tox manages its own test env, then uses pip inside each tox-created env. Up until pip21.3, pip used to create its own build env. But now I think it resorts to the build-in-place feature (which previously available for optional preview to prep pkgs for pip21.3).

---
### Option 1 

This reproduced the problem (even while using a VS dev prompt).

<details><summary>Here's the full error message</summary>

```
PS C:\Users\ytreh\Documents\GitHub\exhale> tox -e py
py create: C:\Users\ytreh\Documents\GitHub\exhale\.tox\py
py installdeps: sphinx, breathe, beautifulsoup4, lxml==4.6.3, six, pytest, pytest-cov, pytest-raises>=0.10, ipdb
ERROR: invocation failed (exit code 1), logfile: C:\Users\ytreh\Documents\GitHub\exhale\.tox\py\log\py-1.log
====================================================== log start =======================================================
Collecting sphinx
  Using cached Sphinx-4.3.1-py3-none-any.whl (3.1 MB)
Collecting breathe
  Using cached breathe-4.31.0-py3-none-any.whl (88 kB)
Collecting beautifulsoup4
  Using cached beautifulsoup4-4.10.0-py3-none-any.whl (97 kB)
Collecting lxml==4.6.3
  Using cached lxml-4.6.3.tar.gz (3.2 MB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting pytest
  Using cached pytest-6.2.5-py3-none-any.whl (280 kB)
Collecting pytest-cov
  Using cached pytest_cov-3.0.0-py3-none-any.whl (20 kB)
Collecting pytest-raises>=0.10
  Using cached pytest_raises-0.11-py2.py3-none-any.whl (9.4 kB)
Collecting ipdb
  Using cached ipdb-0.13.9-py3-none-any.whl
Collecting sphinxcontrib-qthelp
  Using cached sphinxcontrib_qthelp-1.0.3-py2.py3-none-any.whl (90 kB)
Collecting colorama>=0.3.5
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting sphinxcontrib-serializinghtml>=1.1.5
  Using cached sphinxcontrib_serializinghtml-1.1.5-py2.py3-none-any.whl (94 kB)
Collecting requests>=2.5.0
  Using cached requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting Pygments>=2.0
  Using cached Pygments-2.10.0-py3-none-any.whl (1.0 MB)
Requirement already satisfied: setuptools in c:\users\ytreh\documents\github\exhale\.tox\py\lib\site-packages (from sphinx) (58.5.3)
Collecting sphinxcontrib-devhelp
  Using cached sphinxcontrib_devhelp-1.0.2-py2.py3-none-any.whl (84 kB)
Collecting sphinxcontrib-htmlhelp>=2.0.0
  Using cached sphinxcontrib_htmlhelp-2.0.0-py2.py3-none-any.whl (100 kB)
Collecting babel>=1.3
  Using cached Babel-2.9.1-py2.py3-none-any.whl (8.8 MB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting snowballstemmer>=1.1
  Using cached snowballstemmer-2.2.0-py2.py3-none-any.whl (93 kB)
Collecting Jinja2>=2.3
  Using cached Jinja2-3.0.3-py3-none-any.whl (133 kB)
Collecting docutils<0.18,>=0.14
  Using cached docutils-0.17.1-py2.py3-none-any.whl (575 kB)
Collecting imagesize
  Using cached imagesize-1.3.0-py2.py3-none-any.whl (5.2 kB)
Collecting alabaster<0.8,>=0.7
  Using cached alabaster-0.7.12-py2.py3-none-any.whl (14 kB)
Collecting sphinxcontrib-jsmath
  Using cached sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl (5.1 kB)
Collecting sphinxcontrib-applehelp
  Using cached sphinxcontrib_applehelp-1.0.2-py2.py3-none-any.whl (121 kB)
Collecting soupsieve>1.2
  Using cached soupsieve-2.3.1-py3-none-any.whl (37 kB)
Collecting attrs>=19.2.0
  Using cached attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting py>=1.8.2
  Using cached py-1.11.0-py2.py3-none-any.whl (98 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting atomicwrites>=1.0
  Using cached atomicwrites-1.4.0-py2.py3-none-any.whl (6.8 kB)
Collecting coverage[toml]>=5.2.1
  Using cached coverage-6.2-cp310-cp310-win_amd64.whl (183 kB)
Collecting decorator
  Using cached decorator-5.1.0-py3-none-any.whl (9.1 kB)
Collecting ipython>=7.17.0
  Using cached ipython-7.30.1-py3-none-any.whl (791 kB)
Collecting pytz>=2015.7
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting tomli
  Using cached tomli-1.2.2-py3-none-any.whl (12 kB)
Collecting jedi>=0.16
  Using cached jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)
Collecting prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0
  Using cached prompt_toolkit-3.0.24-py3-none-any.whl (374 kB)
Collecting matplotlib-inline
  Using cached matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting traitlets>=4.2
  Using cached traitlets-5.1.1-py3-none-any.whl (102 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.0.1-cp310-cp310-win_amd64.whl (15 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.9-py3-none-any.whl (39 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.6-py3-none-any.whl (97 kB)
Collecting parso<0.9.0,>=0.8.0
  Using cached parso-0.8.3-py2.py3-none-any.whl (100 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Building wheels for collected packages: lxml
  Building wheel for lxml (setup.py): started
  Building wheel for lxml (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ytreh\Documents\GitHub\exhale\.tox\py\Scripts\python.EXE' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ytreh\\AppData\\Local\\Temp\\pip-install-vmksgl5n\\lxml_fe2ee54cda48422aa8f033a1c21f329f\\setup.py'"'"'; __file__='"'"'C:\\Users\\ytreh\\AppData\\Local\\Temp\\pip-install-vmksgl5n\\lxml_fe2ee54cda48422aa8f033a1c21f329f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\ytreh\AppData\Local\Temp\pip-wheel-i572v6bm'
       cwd: C:\Users\ytreh\AppData\Local\Temp\pip-install-vmksgl5n\lxml_fe2ee54cda48422aa8f033a1c21f329f\
  Complete output (94 lines):
  Building lxml version 4.6.3.
  Building without Cython.
  Building against pre-built libxml2 andl libxslt libraries
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win32-3.10
  creating build\lib.win32-3.10\lxml
  copying src\lxml\builder.py -> build\lib.win32-3.10\lxml
  copying src\lxml\cssselect.py -> build\lib.win32-3.10\lxml
  copying src\lxml\doctestcompare.py -> build\lib.win32-3.10\lxml
  copying src\lxml\ElementInclude.py -> build\lib.win32-3.10\lxml
  copying src\lxml\pyclasslookup.py -> build\lib.win32-3.10\lxml
  copying src\lxml\sax.py -> build\lib.win32-3.10\lxml
  copying src\lxml\usedoctest.py -> build\lib.win32-3.10\lxml
  copying src\lxml\_elementpath.py -> build\lib.win32-3.10\lxml
  copying src\lxml\__init__.py -> build\lib.win32-3.10\lxml
  creating build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\__init__.py -> build\lib.win32-3.10\lxml\includes
  creating build\lib.win32-3.10\lxml\html
  copying src\lxml\html\builder.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\clean.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\defs.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\diff.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\ElementSoup.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\formfill.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\html5parser.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\soupparser.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\usedoctest.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\_diffcommand.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\_html5builder.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\_setmixin.py -> build\lib.win32-3.10\lxml\html
  copying src\lxml\html\__init__.py -> build\lib.win32-3.10\lxml\html
  creating build\lib.win32-3.10\lxml\isoschematron
  copying src\lxml\isoschematron\__init__.py -> build\lib.win32-3.10\lxml\isoschematron
  copying src\lxml\etree.h -> build\lib.win32-3.10\lxml
  copying src\lxml\etree_api.h -> build\lib.win32-3.10\lxml
  copying src\lxml\lxml.etree.h -> build\lib.win32-3.10\lxml
  copying src\lxml\lxml.etree_api.h -> build\lib.win32-3.10\lxml
  copying src\lxml\includes\c14n.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\config.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\dtdvalid.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\etreepublic.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\htmlparser.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\relaxng.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\schematron.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\tree.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\uri.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xinclude.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xmlerror.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xmlparser.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xmlschema.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xpath.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\xslt.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\__init__.pxd -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\etree_defs.h -> build\lib.win32-3.10\lxml\includes
  copying src\lxml\includes\lxml-version.h -> build\lib.win32-3.10\lxml\includes
  creating build\lib.win32-3.10\lxml\isoschematron\resources
  creating build\lib.win32-3.10\lxml\isoschematron\resources\rng
  copying src\lxml\isoschematron\resources\rng\iso-schematron.rng -> build\lib.win32-3.10\lxml\isoschematron\resources\rng
  creating build\lib.win32-3.10\lxml\isoschematron\resources\xsl
  copying src\lxml\isoschematron\resources\xsl\RNG2Schtrn.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl
  copying src\lxml\isoschematron\resources\xsl\XSD2Schtrn.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl
  creating build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_abstract_expand.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_dsdl_include.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_schematron_message.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_schematron_skeleton_for_xslt1.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\iso_svrl_for_xslt1.xsl -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  copying src\lxml\isoschematron\resources\xsl\iso-schematron-xslt1\readme.txt -> build\lib.win32-3.10\lxml\isoschematron\resources\xsl\iso-schematron-xslt1
  running build_ext
  building 'lxml.etree' extension
  creating build\temp.win32-3.10
  creating build\temp.win32-3.10\Release
  creating build\temp.win32-3.10\Release\src
  creating build\temp.win32-3.10\Release\src\lxml
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x86\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DCYTHON_CLINE_IN_TRACEBACK=0 -Isrc -Isrc\lxml\includes -IC:\Users\ytreh\Documents\GitHub\exhale\.tox\py\include -IC:\Users\ytreh\AppData\Local\Programs\Python\Python310\include -IC:\Users\ytreh\AppData\Local\Programs\Python\Python310\Include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt /Tcsrc\lxml\etree.c /Fobuild\temp.win32-3.10\Release\src\lxml\etree.obj -w
  cl : Command line warning D9025 : overriding '/W3' with '/w'
  etree.c
  C:\Users\ytreh\AppData\Local\Temp\pip-install-vmksgl5n\lxml_fe2ee54cda48422aa8f033a1c21f329f\src\lxml\includes/etree_defs.h(14): fatal error C1083: Cannot open include file: 'libxml/xmlversion.h': No such file or directory
  Compile failed: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x86\\cl.exe' failed with exit code 2
  creating Users
  creating Users\ytreh
  creating Users\ytreh\AppData
  creating Users\ytreh\AppData\Local
  creating Users\ytreh\AppData\Local\Temp
  C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x86\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -I/usr/include/libxml2 -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt -IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt /TcC:\Users\ytreh\AppData\Local\Temp\xmlXPathInitf810edj1.c /FoUsers\ytreh\AppData\Local\Temp\xmlXPathInitf810edj1.obj
  xmlXPathInitf810edj1.c
  C:\Users\ytreh\AppData\Local\Temp\xmlXPathInitf810edj1.c(1): fatal error C1083: Cannot open include file: 'libxml/xpath.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x86\\cl.exe' failed with exit code 2
  *********************************************************************************
  Could not find function xmlCheckVersion in library libxml2. Is libxml2 installed?
  *********************************************************************************
  ----------------------------------------
  ERROR: Failed building wheel for lxml
  Running setup.py clean for lxml
Failed to build lxml
```
</details>

### Option 2
Using changes suggested in option 1 (specifying `lxml==4.6.3`) and having installed lxml v4.6.3 (from the downloaded wheel) before running tox, the `--sitepackages` arg does help to abate the tox-created env problem. I was able to run `tox  --sitepackages -e py` to completion.

---
> (1) may be worth adding if it is successful, or documenting (2).

I think if you pin tox runs to lxml v4.7.0, there should be no need to document this issue. But, I guess it wouldn't hurt (it is a documentation enhancing software after all 😜).

**Comment by svenevs** [comment]
Awesome, that's super helpful -- thanks!  Will be resolved by #133 which will get merged when lxml finishes their release :slightly_smiling_face: Re: documentation, yeah I think especially having all of the output you included here is very helpful for people searching issues, and anybody trying to develop will likely see the same things and resolutions are discussed here.

**Comment by 2bndy5** [comment]
For the record, I just tried pinning tox to lxml v4.6.4, and this issue didn't occur. You're probably ok to use `lxml>=4.6.4` as v4.6.3 fails in `tox -e py`.
